### ISSUE TITLE
Add simplified filter controls for recipes

### ISSUE THREAD

**Issue by Osmose** [mozilla/normandy#316]
**Blockers**

* [x] #395 - Active filters
* [ ] #396 - Recipe controls

---

We previously stored filter data in the database, and had a built-out form for specifying who recipes should be delivered to. We then switched to using JEXL filter expressions for all filtering, which has a lot of benefits around adding new filters and composing them together. But the UX is still not so good.

We'd like to add some simple filtering back in that gets compiled down to JEXL that is appended to the filter expression. Things we'd like to filter by include:

- Firefox locale
- Geolocated country
- Release channel
- Start and End Time

Along with targetting recipes by these, we'd like to be able to filter by most of these in the recipe listing, and show them in columns in the recipe listing.

**Comment by Osmose** [comment]
I've created a mockup of the UX changes for adding the simple filters: [SHIELD Simple Filters.pdf](https://github.com/mozilla/normandy/files/593123/SHIELD.Simple.Filters.pdf).

**Comment by Osmose** [comment]
@mozilla/strategy-and-insights I'd like to get some feedback on the mockup in the comment above before we start working on it. It's depicts the changes to the recipe listing and recipe form to make common filters easier to enter. The filter expression box will still be available for more complex filters, and can be used in conjunction with simple filters.

What do ya'll think? Any questions?

**Comment by TylerDowner** [comment]
I think it looks great. No specific comments.

**Comment by raymak** [comment]
It looks good. One basic question/comment I have is how can you know that a certain recipe has additional filter expression when you are looking at it in the list?

**Comment by Osmose** [comment]
> It looks good. One basic question/comment I have is how can you know that a certain recipe has additional filter expression when you are looking at it in the list?

Maybe a checkmark/X column flagged on whether the additional filter is empty or not?

**Comment by raymak** [comment]
> It looks good. One basic question/comment I have is how can you know that a certain recipe has additional filter expression when you are looking at it in the list?
> 
> Maybe a checkmark/X column flagged on whether the additional filter is empty or not?

That would be good. Some indication that there's more and this is not all. Then when you click on it it takes you somewhere (preview?) where you can see the additional filters too.

**Comment by jtg567** [comment]
What determines the content of the **Metadata** column? I don't see input controls.

Will it always be empty when action is console-log, and contain **Survey ID:** when action is show-heartbeat?

**Comment by MattGrimes** [comment]
Other than addressing Kamyar's concern, I think it looks great. Very useful.

**Comment by Osmose** [comment]
> What determines the content of the Metadata column? I don't see input controls.
> 
> Will it always be empty when action is console-log, and contain Survey ID: when action is show-heartbeat?

Yep. It's dependent on the recipe type.

In the future when we have many different types of recipes, I suspect we'll have multiple listings, and a survey-specific one would have a survey ID column as opposed to the metadata column.

**Comment by Osmose** [comment]
Here's the final mockups based on feedback: [SHIELD Simple Filters Final.pdf](https://github.com/mozilla/normandy/files/595993/SHIELD.Simple.Filters.Final.pdf)

This should be ready to work on.

**Comment by Osmose** [comment]
Oh yeah, just to be clear: We should be storing these filters in the database, as we did in the past; we used to have a management command that ensured the channels and locales we wanted were in the database and available for foreign keys, which could (or could not) be re-used here. We'll want to pull locale info from `product_details`, and we may want to have a recurring job on prod to pull the latest locale data.

The simple filters should be compiled to JEXL that is included in the filter expression sent to the client; filtering should still happen completely on the client.
